### PR TITLE
Implement IComparable on Rune

### LINKED ISSUE #2340: IComparable for Rune
`Rune` seems to be missing the non-generic `IComparable` interface. Because of this, comparison operators from F# can not be used with the type. Was there a design decision behind omitting this? If not, could this get implemented?

*There's mitigation code down below if you're effected by this in F#. I don't think there's a mitigation for C#/VB.*

### PR DESCRIPTION
**Pull request by GrabYourPitchforks**
Fixes https://github.com/dotnet/runtime/issues/2340.

Adds the `IComparable` interface to `System.Text.Rune`. Follows the same implementation used by `int.CompareTo(object)` and similar methods: _null_ always compares as "less than" _non-null_, and if _obj_ is not null and not a `Rune` an exception is thrown.

### CODE REVIEW

**Comment by GrabYourPitchforks** [comment]
@joperezr Is there an easy way to mark this explicit interface implementation as "just inherit the documentation text from https://docs.microsoft.com/en-us/dotnet/api/system.icomparable.compareto, no need to write anything special"?

**Comment by safern** [comment]
cc: @billwert @gewarren @carlossanlop might know the answer to the question above

**Comment by joperezr** [comment]
yeah to be honest I'm not sure, I suppose you tried `///<inheritdoc cref="IComparable"/>` ? That is supposed to work at least with regular projects but the fact that we use special infra for docs might make that not work, @carlossanlop might know.

File: src/libraries/System.Runtime/ref/System.Runtime.cs [hunk 1]
```diff
@@ -10565,7 +10565,7 @@ public enum NormalizationForm
         FormKC = 5,
         FormKD = 6,
     }
-    public readonly partial struct Rune : System.IComparable<System.Text.Rune>, System.IEquatable<System.Text.Rune>
+    public readonly partial struct Rune : System.IComparable, System.IComparable<System.Text.Rune>, System.IEquatable<System.Text.Rune>
```
Comment (safern) [inline src/libraries/System.Runtime/ref/System.Runtime.cs:10568]: This seems to be missing the implementation of `CompareTo(Rune)`

Comment (GrabYourPitchforks) [reply src/libraries/System.Runtime/ref/System.Runtime.cs:10568]: It's there, just a bit hidden. :)
The existing generic `CompareTo(Rune)` API is implicitly implemented so it's in alphabetical order with everything else.
The new non-generic `CompareTo(object)` API is explicitly implemented so it's at the end of the type declaration.

Comment (safern) [reply src/libraries/System.Runtime/ref/System.Runtime.cs:10568]: Ah I see. I clicked on the expand lines in the UI but given the size of this file it didn't show it. Looking at it in raw form now I can see it 😄 

This is what GH was showing me:
image

🤦

**Comment by GrabYourPitchforks** [comment]
Put the `<inheritdoc>` marker on it for now per Joe's suggestion. If Carlos has other suggestions I'll merge those in. Thanks!

**Review by safern — APPROVED** [review]

**Comment by joperezr** [comment]
> Put the <inheritdoc> marker on it for now per Joe's suggestion.

That seemed to have worked:
```xml
        <member name="M:System.Text.Rune.System#IComparable#CompareTo(System.Object)">
            <inheritdoc cref="M:System.IComparable.CompareTo(System.Object)" />
        </member>
```

**Review by joperezr — APPROVED** [review]

**Comment by safern** [comment]
We will most likely need to port the docs for the new API to dotnet/docs though.

File: src/libraries/System.Private.CoreLib/src/System/Text/Rune.cs [hunk 2]
```diff
@@ -269,7 +269,13 @@ private static Rune ChangeCaseCultureAware(Rune rune, CultureInfo culture, bool
         }
 #endif
 
-        public int CompareTo(Rune other) => _value.CompareTo(other._value);
+        public int CompareTo(Rune other)
```
Comment (tarekgh) [inline src/libraries/System.Private.CoreLib/src/System/Text/Rune.cs:272]: nit: use the => syntax and keep the comment too.

**Review by tarekgh — APPROVED** [review]

**Pull request merged**